spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/messaging/TracingMethodMessageHandlerAdapter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.messaging;
+
+import java.util.function.Consumer;
+
+import brave.Span;
+import brave.Tracer;
+import brave.Tracing;
+import brave.propagation.Propagation;
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContextOrSamplingFlags;
+
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageHeaderAccessor;
+
+public class TracingMethodMessageHandlerAdapter {
+
+	private Tracer tracer;
+
+	private TraceContext.Extractor<MessageHeaderAccessor> extractor;
+
+	public TracingMethodMessageHandlerAdapter(Tracing tracing,
+			Propagation.Getter<MessageHeaderAccessor, String> traceMessagePropagationGetter) {
+		this.tracer = tracing.tracer();
+		this.extractor = tracing.propagation().extractor(traceMessagePropagationGetter);
+	}
+
+	public void wrapMethodMessageHandler(Message<?> message,
+			Consumer<Message<?>> methodMessageHandler) {
+		TraceContextOrSamplingFlags extracted = extractor
```

Comment:
when extracting a message, we should also clear the properties like this https://github.com/openzipkin/brave/blob/master/instrumentation/spring-rabbit/src/main/java/brave/spring/rabbit/SpringRabbitTracing.java#L194